src/main/java/org/jabref/gui/maintable/MainTableColumnFactory.java

```diff
@@ -269,20 +277,50 @@ private ContextMenu createFileMenu(BibEntryTableViewModel entry, List<LinkedFile
     }
 
     /**
-     * Creates a column which shows an icon instead of the textual content
+     * Creates a column for DOIs or URLs.
+     * The {@code firstField} is preferred to be shown over {@code secondField}.
      */
-    private TableColumn<BibEntryTableViewModel, String> createIconColumn(JabRefIcon icon, String firstField, String secondField) {
+    private TableColumn<BibEntryTableViewModel, String> createUrlOrDoiColumn(JabRefIcon icon, String firstField, String secondField) {
         TableColumn<BibEntryTableViewModel, String> column = new TableColumn<>();
         column.setGraphic(icon.getGraphicNode());
         column.getStyleClass().add(ICON_COLUMN);
         setExactWidth(column, GUIGlobals.WIDTH_ICON_COL);
-        column.setCellValueFactory(cellData -> EasyBind.monadic(cellData.getValue().getField(firstField)).orElse(cellData.getValue().getField(secondField)));
+        column.setCellValueFactory(cellData -> EasyBind.monadic(cellData.getValue().getField(firstField)).map(x -> firstField).orElse(EasyBind.monadic(cellData.getValue().getField(secondField)).map(x -> secondField)));
                 new ValueTableCellFactory<BibEntryTableViewModel, String>()
                         .withGraphic(cellFactory::getTableIcon)
+                        .withOnMouseClickedEvent((entry, content) -> createOpenUrlOrDoiHandler(entry, firstField, secondField))
                         .install(column);
         return column;
     }
 
+    private EventHandler<? super MouseEvent> createOpenUrlOrDoiHandler(BibEntryTableViewModel entry, String firstField, String secondField) {
+        return (event) -> {
+            if (event.getButton() != MouseButton.PRIMARY) {
+                return;
+            }
+
+            String field;
+            if (entry.getEntry().hasField(firstField)) {
+                field = firstField;
+            } else if (entry.getEntry().hasField(secondField)) {
+                field = secondField;
+            } else {
+                // nothing to open
+                LOGGER.debug("Requested opening DOI or URL, but none present.");
+                return;
+            }
+
+            entry.getEntry().getField(field).ifPresent(url -> {
+                try {
+                    JabRefDesktop.openExternalViewer(database, url, field);
+                } catch (IOException e) {
+                    LOGGER.error("Failed to open external viewer for '{}'", url, e);
+                }
+            });
+            event.consume();
+        };
+    }
+
     private TableColumn<BibEntryTableViewModel, String> createIconColumn(JabRefIcon icon, String field) {
```

Comment:
Could you please fix this method for the `eprint` column as well. Thanks!

Reply:
Done!